elide-core/src/main/java/com/yahoo/elide/utils/coerce/CoerceUtil.java

```diff
@@ -70,6 +74,9 @@ public Converter lookup(Class<?> sourceType, Class<?> targetType) {
                     return TO_ENUM_CONVERTER;
                 } else if (Map.class.isAssignableFrom(sourceType)) {
                     return FROM_MAP_CONVERTER;
+                } else if ((String.class.isAssignableFrom(sourceType) || Long.class.isAssignableFrom(sourceType))
```

Comment:
@xiaoyao1991  @DeathByTape  I've found a bug here. If you try to create entity with date field and set value to 1970-01-01 (timestamp is -14400000) it will be automactically converted into `Integer` object, not `Long` inside elide because the value is too small and it seems like the smallest allowed number type is selected on conversion. Then it will miss this if branch and wrong date converter will be chosen.